repose-aggregator/components/filters/uri-stripper-filter/src/main/scala/org/openrepose/filters/uristripper/UriStripperFilter.scala

```diff
@@ -97,34 +107,129 @@ class UriStripperFilter @Inject()(configurationService: ConfigurationService)
+        .filter(resource => isMatchingMethod(resource.getHttpMethods.toSet, wrappedRequest.getMethod))
+        .map(resource => getPathsForContentType(wrappedResponse.getContentType, resource))
+        .fold(List.empty[LinkPath])(_ ++ _)
+
+      if (token.isDefined && applicableLinkPaths.nonEmpty) {
+        val tryParsedJson = Try(Json.parse(wrappedResponse.getOutputStreamAsInputStream))
+        applicableLinkPaths.foldLeft(tryParsedJson)(transformLink(_, _, token.get, previousToken, nextToken)) match {
+          case Success(jsValue) =>
+            val jsValueBytes = jsValue.toString.getBytes(wrappedResponse.getCharacterEncoding)
+            wrappedResponse.setContentLength(jsValueBytes.length)
+            wrappedResponse.setOutput(new ByteArrayInputStream(jsValueBytes))
+          case Failure(e) =>
+            wrappedResponse.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, e.getMessage)
+        }
+      }
+
       wrappedResponse.commitToResponse()
     }
   }
 
-  private def splitUriIntoTokens(uri: String): mutable.Buffer[String] =
+  private def splitPathIntoTokens(uri: String): mutable.Buffer[String] =
     StringUriUtilities.formatUriNoLead(uri).split(UriDelimiter).toBuffer
 
-  private def joinTokensIntoUri(tokens: mutable.Buffer[String]): String =
+  private def joinTokensIntoPath(tokens: mutable.Buffer[String]): String =
     StringUriUtilities.formatUri(tokens.mkString(UriDelimiter))
 
+  private def findReplacementTokenIndex(pathTokens: mutable.Buffer[String], previousToken: Option[String], nextToken: Option[String]): Option[Int] = {
+    (previousToken.filter(pathTokens.contains), nextToken.filter(pathTokens.contains)) match {
+      case (Some(foundToken), _) => Some(pathTokens.indexOf(foundToken) + 1)
+      case (_, Some(foundToken)) => Some(pathTokens.indexOf(foundToken))
+      case _ => None
+    }
+  }
+
+  private def replacePath(uri: URI, newPath: String): String = {
+    val preUri = Option(uri.getScheme).map(_ + "://" + uri.getHost + (if (uri.getPort != -1) ":" + uri.getPort else ""))
+    val postUri = Option(uri.getQuery).filterNot(_.trim.isEmpty).map(QueryParamIndicator +)
+    preUri.getOrElse("") + newPath + postUri.getOrElse("")
+  }
+
+  private def isMatchingMethod(configuredMethods: Set[HttpMethod], requestMethod: String): Boolean =
+    configuredMethods.isEmpty ||
+      configuredMethods.contains(HttpMethod.ALL) ||
+      configuredMethods.contains(HttpMethod.fromValue(requestMethod))
+
+  private def getPathsForContentType(contentType: String, resource: LinkResource): List[LinkPath] = {
+    Option(contentType) match {
+      case Some(ct) if ct.toLowerCase.contains("json") => resource.getJson.toList
+      // todo: return xpath when xml is supported
+      case Some(ct) if ct.toLowerCase.contains("xml") => List.empty
+      case _ => List.empty
+    }
+  }
+
+  // todo: this function only provides support for named '$' and '.' JSONPath operators
```

Comment:
The other library (https://github.com/josephpconley/play-jsonpath) really only helps retrieve a value given a JSONPath. Unfortunately, that doesn't help us transform the input JSON (the `JsValue` returned by the other library will not have a reference to its parent). To transform the JSON, we could use a combination of the other library with a JSON Patch library, but then things start to get a bit complex.

I think the ideal solution would be to find a way to map the library agnostic typing from the Gatling JSONPath parser to a Play JSON `JsPath`. The `parse` function from the other library does something somewhat similar, and may be helpful in figuring out what to do.